flink/v1.16/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -390,4 +393,40 @@ public void testOverrideWriteConfigWithUnknownFileFormat() {
           return null;
         });
   }
+
+  @Test
+  public void testPartitionEvolution() throws Exception {
```

Comment:
On second thought, I don't think this UT is stable enough to test your case.  My test also confirmed my point, I ran the UT many times, sometimes it passed, sometimes it didn't.

I think we should use `TestIcebergFilesCommitter` to test it. Specifically, we can change its table partition between commits to validate your code.

Reply:
+1, use multi-thread in unit test will cause unstable result.

Reply:
Updated to `TestIcebergFilesCommitter `. @hililiwei which errors are you met? I have run several times and all are passed.

Reply:
Sorry for the late reply. The main problem is `multi-thread`, sometimes it doesn't work as expected, but I can't remember the details.

Reply:
Thanks for the reply, the UTs have updated to `TestIcebergFilesCommitter `.